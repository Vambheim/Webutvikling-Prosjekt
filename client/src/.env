# Måte å tilby kryptert API-nøkkel (gir tryggere kommunikasjon mot nettet) 
REACT_APP_API_KEY=5b7869e75f75481a8f5005bc7bffd3a8 
