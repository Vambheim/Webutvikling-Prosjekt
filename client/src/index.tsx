import ReactDOM from 'react-dom';
import * as React from 'react';
import { Component } from 'react-simplified';
import { HashRouter, Route } from 'react-router-dom';
import { Alert, Card } from './widgets';
import {
  RecipeAdd,
  RecipeList,
  ShoppingList,
  TaskDetails,
  TaskEdit,
  TaskNew,
} from './recipe-components';
import Container from 'react-bootstrap/Container';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import NavDropdown from 'react-bootstrap/NavDropdown';
// import Card from 'react-bootstrap/Card';

class Menu extends Component {
  render() {
    return (
      <Navbar bg="light" expand="lg">
        <Container>
          <Navbar.Brand href="/" className="ms-auto">
            Java matapplikasjon
          </Navbar.Brand>
          <Navbar.Toggle aria-controls="basic-navbar-nav" />
          <Navbar.Collapse id="basic-navbar-nav">
            <Nav className="ms-auto">
              <Nav.Link href="/#/recipes">Oppskrifter</Nav.Link>
              <Nav.Link href="/#/recipes/add">Legg til oppskrift</Nav.Link>
              <Nav.Link href="/#/recipes/cart">Handleliste</Nav.Link>
              <NavDropdown title="Innstillinger" id="basic-nav-dropdown">
                <NavDropdown.Item href="/#/recipes/cart">Profil</NavDropdown.Item>
                <NavDropdown.Item href="#action/3.2">MÃ¸rk modus</NavDropdown.Item>
                <NavDropdown.Divider />
                <NavDropdown.Item href="#action/3.4">Overrask meg</NavDropdown.Item>
              </NavDropdown>
            </Nav>
          </Navbar.Collapse>
        </Container>
      </Navbar>
    );
  }
}

class Home extends Component {
  render() {
    return <Card title="Welcome">This is your favourite food recipe app</Card>;
  }
}

ReactDOM.render(
  <HashRouter>
    <div>
      <Alert />
      <Menu />
      <Route exact path="/" component={Home} />
      <Route exact path="/recipes" component={RecipeList} />
      <Route exact path="/recipes/add" component={RecipeAdd} />
      <Route exact path="/recipes/cart" component={ShoppingList} />
      <Route exact path="/tasks/:id(\d+)" component={TaskDetails} /> {/* id must be number */}
      <Route exact path="/tasks/:id(\d+)/edit" component={TaskEdit} /> {/* id must be number */}
      <Route exact path="/tasks/new" component={TaskNew} />
    </div>
  </HashRouter>,
  document.getElementById('root')
);
