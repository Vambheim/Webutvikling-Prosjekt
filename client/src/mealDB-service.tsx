import axios from 'axios';

axios.defaults.baseURL = 'https://www.themealdb.com/api/json/v1/1';

export type MealDBRecipe = {
  mealDBId: number;
  name: string;
  category: string;
  country: string;
};

class MealDBService {
  /**
   * Get task with given id.
   */
  get(mealDBId: number) {
    return axios.get<MealDBRecipe>('/lookup.php?i=' + mealDBId).then((response) => response.data);
  }

  /**
   * Get all tasks.
   */
  // getAll() {
  //   return axios.get<Recipe[]>('/recipes').then((response) => response.data);
  // }

  /**
   * Create new task having the given title.
   *
   * Resolves the newly created task id.
   */
  // create(title: string) {
  //   return axios
  //     .post<{ id: number }>('/tasks', { title: title })
  //     .then((response) => response.data.id);
  // }
}

const mealDBService = new MealDBService();
export default mealDBService;
