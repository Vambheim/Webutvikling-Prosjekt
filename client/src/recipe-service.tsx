import axios from 'axios';

axios.defaults.baseURL = 'http://localhost:3000/api/v2';

export type Recipe = {
  recipe_id: number;
  name: string;
  category: string;
  country: string;
};

export type RecipeDetailed = {
  recipe_id: number;
  name: string;
  category: string;
  country: string;
  ingriedients: Array<Ingredient>;
};

export type Step = {
  step_id: number;
  order_number: number;
  description: string;
  recipe_id: number;
};

export type RecipeIngredient = {
  ingredient_id: number;
  name: string;
  recipe_id: number;
  amount_per_person: number;
  measurement_unit: string;
};

export type Ingredient = {
  ingredient_id: number;
  name: string;
};

export type addIngredient = {
  name: string;
  amount: number;
  measurement_unit: string;
};

export type addStep = {
  description: string;
  order_number: number;
};

class RecipeService {
  /**
   * Get recipe with given recipe_id.
   */
  get(recipe_id: number) {
    return axios.get<Recipe>('/recipes/' + recipe_id).then((response) => response.data);
  }

  /**
   * Get all recipes.
   */
  getAll() {
    return axios.get<Recipe[]>('/recipes').then((response) => response.data);
  }

  /**
   * Create new recipe.
   *
   * Resolves the newly created task id.
   */
  createRecipe(name: string, category: string, country: string) {
    return axios
      .post<{ recipe_id: number }>('/recipes', {
        name: name,
        category: category,
        country: country,
      })
      .then((response) => response.data.recipe_id);
  }

  /**
   * Updates given recipe
   */
  update(recipe: Recipe) {
    return axios.put('/recipes', recipe).then((response) => response.data);
  }

  /**
   * Delete recipe with given id
   */
  delete(recipe_id: number) {
    return axios.delete('/recipes/' + recipe_id).then((response) => response.data);
  }

  /**
   * Get ingredients with given recipe_id
   */
  getRecipeIngredients(recipe_id: number) {
    return axios
      .get<RecipeIngredient[]>('/recipes/' + recipe_id + '/ingredients')
      .then((response) => response.data);
  }

  /**
   * Get all ingredients in the database
   */
  getAllIngredients() {
    return axios.get<Ingredient[]>('/ingredients').then((response) => response.data);
  }

  /**
   * Creates connection with recipe and ingredient
   */
  createRecipeIngredients(
    name: string,
    recipe_id: number,
    amount_per_person: number,
    measurement_unit: string
  ) {
    return axios
      .post('/recipes/ingredients', {
        name: name,
        recipe_id: recipe_id,
        amount_per_person: amount_per_person,
        measurement_unit: measurement_unit,
      })
      .then((response) => response.data);
  }

  /**
   * Updates given ingredient in given recipe
   */
  updateRecipeIngredient(
    amount_per_person: number,
    measurement_unit: string,
    recipe_id: number,
    ingredient_id: number,
    name: string
  ) {
    return axios
      .put('/recipes/' + recipe_id + '/ingredients/' + ingredient_id, {
        amount_per_person: amount_per_person,
        measurement_unit: measurement_unit,
        name: name,
      })
      .then((response) => response.data);
  }

  /**
   * Get all steps with given recipe_id
   */
  getSteps(recipe_id: number) {
    return axios.get<Step[]>('/recipes/' + recipe_id + '/steps').then((response) => response.data);
  }

  /**
   * Creates new step to given recipe
   */
  createStep(order_number: number, description: string, recipe_id: number) {
    return axios
      .post('/steps', {
        order_number: order_number,
        description: description,
        recipe_id: recipe_id,
      })
      .then((response) => response.data);
  }

  /**
   * Updates given step in given recipe
   */
  updateStep(recipe_id: number, step_id: number, order_number: number, description: string) {
    return axios
      .put('recipes/' + recipe_id + '/steps/' + step_id, {
        order_number: order_number,
        description: description,
      })
      .then((response) => response.data);
  }

  /**
   * Get recipes filtered by country and category
   */
  getFilterByCountryAndCategory(country: string, category: string) {
    return axios
      .get<Recipe[]>('/countryandcategoryfilter/' + country + '/' + category)
      .then((response) => response.data);
  }

  /**
   * Get recipes filtered by category
   */
  getFilterByCategory(category: string) {
    return axios.get<Recipe[]>('/categoryfilter/' + category).then((response) => response.data);
  }

  /**
   * Get recipes filtered by country
   */
  getFilterByCountry(country: string) {
    return axios.get<Recipe[]>('/countryfilter/' + country).then((response) => response.data);
  }

  /**
   * Get recipes filtered by one ingredient
   */
  getFilterByOneIngredient(ingredient1: string) {
    return axios
      .get<Recipe[]>('/oneingredientfilter/' + ingredient1)
      .then((response) => response.data);
  }

  /**
   * Get recipes filtered by two ingredients
   */
  getFilterBy2Ingredients(ingredient1: string, ingredient2: string) {
    return axios
      .get<Recipe[]>('/twoingredientsfilter/' + ingredient1 + '/' + ingredient2)
      .then((response) => response.data);
  }

  /**
   * Get recipes filtered by three ingredients
   */
  getFilterBy3Ingredients(ingredient1: string, ingredient2: string, ingredient3: string) {
    return axios
      .get<Recipe[]>(
        '/threeingredientsfilter/' + ingredient1 + '/' + ingredient2 + '/' + ingredient3
      )
      .then((response) => response.data);
  }

  /**
   * Get recommended recipes to given recipe
   */
  getRecommendedRecipes(recipe_id: number, category: string, country: string) {
    return axios
      .get<Recipe[]>('/recipes/' + recipe_id + '/recommended/' + category + '/' + country)
      .then((response) => response.data);
  }

  /**
   * Likes a given recipe from given user
   */
  likeRecipe(user_id: number, recipe_id: number) {
    return axios
      .post('/recipes/like', { user_id: user_id, recipe_id: recipe_id })
      .then((response) => response.data);
  }

  /**
   * Get liked recipes
   */
  getLikedRecipes(user_id: number) {
    return axios.get<Recipe[]>('/likedRecipes/' + user_id).then((response) => response.data);
  }

  /**
   * Posts recipes from spoonacular
   */
  PostSpoonacularRecipes(recipes: Array<RecipeDetailed>) {
    return axios
      .post<Array<RecipeDetailed>>('/spoonacular/recipes/', { recipes })
      .then((response) => response.data);
  }

  /**
   * Posts ingredients from spoonacular
   */
  PostSpoonacularIngriedents(ingridients: Array<RecipeIngredient>) {
    return axios
      .post<Array<RecipeIngredient>>('/spoonacular/ingridients/', { ingridients })
      .then((response) => response.data);
  }

  /**
   * Posts connection with ingredient and recipe from spoonacular
   */
  PostSpoonacularRecipeIngriedents(data: Array<RecipeDetailed>) {
    var ingridients = [];

    for (let i = 0; i < data.length;) {
      ingridients.push(data[i].ingriedients);
      i++;
    }
    ingridients = ingridients.flat();

    return axios
      .post<Array<RecipeIngredient>>('/spoonacular/ingridients-recipes/', { ingridients })
      .then((response) => response.data);
  }

  /**
   * Posts steps from spoonacular
   */
  PostSpoonacularSteps(steps: Array<Step>) {
    return axios
      .post<Array<Step>>('/spoonacular/steps/', { steps })
      .then((response) => response.data);
  }
}

const recipeService = new RecipeService();
export default recipeService;
