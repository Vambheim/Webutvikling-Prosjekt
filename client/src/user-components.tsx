import * as React from 'react';
import { Component } from 'react-simplified';
import { Alert, Column } from './widgets';
import { Button, Form, Card, Row, Col, Container } from 'react-bootstrap';
import { NavLink } from 'react-router-dom';
import recipeService, { Recipe } from './recipe-service';
import userService, { User } from './user-service';
import { createHashHistory } from 'history';

//false as default
export let loggedIn: boolean = false;
export let currentUser: User = {
  user_id: 0,
  email: '',
  first_name: '',
  last_name: '',
  password: '',
};

const history = createHashHistory(); // Use history.push(...) to programmatically change path, for instance after successfully saving a recipe

export class UserLogIn extends Component {
  email: string = '';
  password: string = '';

  render() {
    return (
      <Card
        style={{
          border: 'none',
          padding: '15px',
          textAlign: 'center',
          marginLeft: 'auto',
          marginRight: 'auto',
        }}
      >
        {/*Card forms in for log in screen */}
        <Card.Title>Log in</Card.Title>
        <Container style={{ width: '20rem', marginLeft: 'auto', marginRight: 'auto' }}>
          <Row>
            <Form.Control
              value={this.email}
              type="text"
              placeholder="Email"
              onChange={(event) => (this.email = event.currentTarget.value)}
              style={{
                textAlign: 'center',
                marginBottom: '10px',
              }}
            ></Form.Control>
          </Row>
          <Row>
            <Form.Control
              value={this.password}
              type="password"
              placeholder="Password"
              onChange={(event) => (this.password = event.currentTarget.value)}
              // Makes it possible to log in with enter as well as with button
              onKeyUp={(event) => {
                if (event.key == 'Enter') {
                  this.logIn();
                }
              }}
              style={{
                textAlign: 'center',
                marginBottom: '10px',
              }}
            ></Form.Control>
          </Row>
        </Container>
        {/*Card for buttons in login screen before user is identified or registered */}
        <Container style={{ width: '15rem', marginLeft: 'auto', marginRight: 'auto' }}>
          <Row>
            <Button
              variant="success"
              onClick={() => this.logIn()}
              style={{
                marginBottom: '10px',
              }}
            >
              Log in
            </Button>
          </Row>
          <Row>
            <Button
              variant="outline-success"
              onClick={() => this.createUser()}
              style={{
                marginBottom: '10px',
              }}
            >
              No user? Create one here
            </Button>
          </Row>
          <Row>
            <Button
              variant="outline-success"
              onClick={() => this.clearInput()}
              style={{
                marginBottom: '10px',
              }}
            >
              Clear input
            </Button>
          </Row>
        </Container>
      </Card>
    );
  }

  logIn() {
    if (this.email.length != 0 && this.password.length != 0) {
      userService
        .logIn(this.email, this.password)
        .then((user) => {
          currentUser = user;
          loggedIn = true;
          Alert.success('Logged in as ' + currentUser.email);
          history.push('/recipes/user');
        })
        .catch((error) => Alert.danger(error.response.data));
    } else {
      Alert.danger('Please fill in all the fields');
    }
  }

  clearInput() {
    this.email = '';
    this.password = '';
  }

  createUser() {
    history.push('/recipes/register');
  }
}

export class RegisterUser extends Component {
  user: User = { user_id: 0, email: '', first_name: '', last_name: '', password: '' };
  confirm_password: string = '';

  render() {
    return (
      <Card
        style={{
          border: 'none',
          padding: '15px',
          textAlign: 'center',
          marginLeft: 'auto',
          marginRight: 'auto',
        }}
      >
        {/* Card creating forms related to creating new user */}
        <Card.Title>Create user</Card.Title>
        <Container
          style={{
            marginLeft: 'auto',
            marginRight: 'auto',
            width: '20rem',
          }}
        >
          <Row>
            <Form.Control
              value={this.user.email}
              type="text"
              placeholder="Email"
              onChange={(event) => (this.user.email = event.currentTarget.value)}
              style={{
                marginBottom: '10px',
                textAlign: 'center',
              }}
            ></Form.Control>
          </Row>
          <Row>
            <Form.Control
              value={this.user.first_name}
              type="text"
              placeholder="First name"
              onChange={(event) => (this.user.first_name = event.currentTarget.value)}
              style={{
                marginBottom: '10px',
                textAlign: 'center',
              }}
            ></Form.Control>
          </Row>
          <Row>
            <Form.Control
              value={this.user.last_name}
              type="text"
              placeholder="Last name"
              onChange={(event) => (this.user.last_name = event.currentTarget.value)}
              style={{
                marginBottom: '10px',
                textAlign: 'center',
              }}
            ></Form.Control>
          </Row>
          <Row>
            <Form.Control
              value={this.user.password}
              type="password"
              placeholder="Password"
              onChange={(event) => (this.user.password = event.currentTarget.value)}
              // Makes it possible to log in with enter as well as with button
              onKeyUp={(event) => {
                if (event.key == 'Enter') {
                  this.createUser();
                }
              }}
              style={{
                marginBottom: '10px',
                textAlign: 'center',
              }}
            ></Form.Control>
          </Row>
          <Row>
            <Form.Control
              value={this.confirm_password}
              type="password"
              placeholder="Confirm password"
              onChange={(event) => (this.confirm_password = event.currentTarget.value)}
              onKeyUp={(event) => {
                if (event.key == 'Enter') {
                  this.createUser();
                }
              }}
              style={{
                marginBottom: '10px',
                textAlign: 'center',
              }}
            ></Form.Control>
          </Row>
        </Container>
        {/* Buttons for creating user and clearing input */}
        <Container style={{ width: '15rem', marginLeft: 'auto', marginRight: 'auto' }}>
          <Row>
            <Button
              variant="success"
              onClick={() => this.createUser()}
              style={{
                marginBottom: '10px',
              }}
            >
              Create user
            </Button>
          </Row>
          <Row>
            <Button
              variant="outline-success"
              onClick={() => this.clearInput()}
              style={{
                marginBottom: '10px',
              }}
            >
              Clear input
            </Button>
          </Row>
        </Container>
      </Card>
    );
  }

  createUser() {
    userService
      .createUser(
        this.user.email,
        this.user.first_name,
        this.user.last_name,
        this.user.password,
        this.confirm_password
      )
      .then((response) => {
        if (response.length > 0) {
          Alert.danger(response);
        } else {
          Alert.success('User created, please log in');
          loggedIn = true;
          history.push('/recipes/login');
        }
      })
      .catch((error) => Alert.danger(error.response.data));
  }

  clearInput() {
    this.user = { user_id: 0, email: '', first_name: '', last_name: '', password: '' };
    this.confirm_password = '';
  }
}

export class UserDetails extends Component {
  likedRecipes: Recipe[] = [];
  render() {
    return (
      <>
        <Card
          style={{
            // border: 'none',
            padding: '15px',
            textAlign: 'center',
            marginLeft: 'auto',
            marginRight: 'auto',
          }}
        >
          {/* Page for all relevant user info for logged in user */}
          <Card.Title>
            {'User page for ' + currentUser.first_name + ' ' + currentUser.last_name}
          </Card.Title>
          <Row style={{ fontSize: '17px' }}>
            <Card.Text>Your email-adress: {currentUser.email}</Card.Text>
          </Row>
          <Row style={{ fontSize: '17px' }}>
            <Card.Text>
              Your name: {currentUser.first_name} {currentUser.last_name}
            </Card.Text>
          </Row>
          <Row>
            <Button
              variant="outline-danger"
              onClick={() => this.logOut()}
              style={{
                width: '15rem',
                marginLeft: 'auto',
                marginRight: 'auto',
                marginBottom: '10px',
              }}
            >
              Log out
            </Button>
          </Row>
        </Card>

        {/* Card retrieving active users liked recipes */}
        <Card
          style={{
            border: 'none',
            padding: '15px',
            textAlign: 'center',
            marginLeft: 'auto',
            marginRight: 'auto',
          }}
        >
          <Card.Title>Your liked recipes</Card.Title>
          <Row>
            <Col lg>
              <Row xs={1} md={4} className="g-4">
                {this.likedRecipes.map((recipe) => (
                  <NavLink
                    to={'/recipes/' + recipe.recipe_id}
                    style={{
                      color: '#9FC1C0',
                    }}
                  >
                    <Column>
                      <Card
                        style={{
                          width: '100%',
                          margin: '1%',

                          textAlign: 'center',
                          borderLeft: 'none',
                          borderRight: 'none',
                          borderTop: 'none',
                          borderRadius: 'none',
                        }}
                      >
                        <Card.Img variant="top" src="https://s.tihlde.org/recipechef12312" />
                        <Card.Body>
                          <Card.Title style={{ color: 'rgb(82, 130, 101)' }}>
                            {recipe.name}
                          </Card.Title>
                          <Card.Text>
                            {recipe.country} {recipe.category}
                          </Card.Text>
                        </Card.Body>
                      </Card>
                    </Column>
                  </NavLink>
                ))}
              </Row>
            </Col>
          </Row>
        </Card>
      </>
    );
  }

  mounted() {
    if (!loggedIn) {
      history.push('/recipes/login');
    } else {
      recipeService
        .getLikedRecipes(currentUser.user_id)
        .then((recipes) => (this.likedRecipes = recipes))
        .catch((error) => Alert.danger(error.message));
    }
  }

  logOut() {
    loggedIn = false;
    history.push('/recipes');
    currentUser = { user_id: 0, email: '', first_name: '', last_name: '', password: '' };
  }
}
