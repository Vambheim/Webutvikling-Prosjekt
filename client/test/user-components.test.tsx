import * as React from 'react';
import { UserLogIn, RegisterUser, UserDetails } from '../src/user-components';
import { shallow } from 'enzyme';
import { NavLink } from 'react-router-dom';
import { Component } from 'react-simplified';
import { Alert, Column } from '../src/widgets';
import { Button, Form, Card, Row, Col, Container, FormControl } from 'react-bootstrap';
import recipeService, { Recipe } from '../src/recipe-service';
import userService, { User } from '../src/user-service';
import { createHashHistory } from 'history';

jest.mock('../src/user-service', () => {
  class UserService {
    createUser(
      email: 'test@mail.com',
      first_name: 'testFirstName',
      last_name: 'testLastName',
      password: 'testPassword'
    ) {
      return Promise.resolve({
        user_id: 10000,
        email: 'test@mail.com',
        first_name: 'testFirstName',
        last_name: 'testLastName',
        password: 'testPassword',
      });
    }

    logIn(email: 'test@mail.com', password: 'testPassword') {
      return Promise.resolve({
        user_id: 1,
        email: 'test@mail.com',
        first_name: 'testFirstName',
        last_name: 'testLastName',
        password: 'testPassword',
      });
    }
  }
  return new UserService();
});

jest.mock('../src/recipe-service', () => {
  class RecipeService {
    getLikedRecipes(user_id: 1) {
      return Promise.resolve([
        {
          recipe_id: 631748,
          name: 'Asian Shrimp Stir-Fry',
          category: 'lunch',
          country: 'Asian',
        },
        {
          recipe_id: 631807,
          name: 'Toasted" Agnolotti (or Ravioli)',
          category: 'side dish',
          country: 'Mediterranean',
        },
      ]);
    }
  }
  return new RecipeService();
});

// describe('UserLogIn component tests', () => {
//   test('UserLogIn draws correctly', () => {
//     const wrapper = shallow(<UserLogIn />);

//     expect(wrapper).toMatchSnapshot();
//   });
// });

describe('RegisterUser component tests', () => {
  test('RegisterUser draws correctly', () => {
    const wrapper = shallow(<RegisterUser />);

    expect(wrapper).toMatchSnapshot();
  });
});

describe('UserDetails component tests', () => {
  test('UserDetails draws correctly', () => {
    const wrapper = shallow(<UserDetails />);

    expect(wrapper).toMatchSnapshot();
  });
});

describe('UserLogIn widgets tests', () => {
  test('UserLogIn draws correctly', () => {
    const wrapper = shallow(<UserLogIn />);

    expect(wrapper).toMatchSnapshot();
  });
  test('Input updates correctly', () => {
    const wrapper = shallow(<UserLogIn />);

    // @ts-ignore: do not type check next line.
    expect(wrapper.containsMatchingElement(<Form.Control value="" />)).toEqual(true);

    wrapper
      .find(Form.Control)
      .at(0)
      .simulate('change', { currentTarget: { value: 'test1' } });

    wrapper
      .find(Form.Control)
      .at(1)
      .simulate('change', { currentTarget: { value: 'test2' } });

    // @ts-ignore: do not type check next line.
    expect(wrapper.containsMatchingElement(<Form.Control value="test1" />)).toEqual(true);
    expect(wrapper.containsMatchingElement(<Form.Control value="test2" />)).toEqual(true);
  });

  // test('Button sucess calls function on click-event', () => {
  //   const wrapper = shallow(
  //     <Button variant="success" onClick={() => userService.logIn('test@mail.com', 'testPassword')}>
  //       Log in
  //     </Button>
  //   );

  //   wrapper.find('button').simulate('click');

  //   expect().toEqual('Logged in as test@mail.com');
  // });
});

// describe('Button.Success widget tests', () => {
//   test('Draws correctly', () => {
//     // @ts-ignore: do not type check next line.
//     const wrapper = shallow(<Button.Success>test</Button.Success>);

//     expect(
//       wrapper.matchesElement(
//         <button type="button" className="btn btn-success">
//           test
//         </button>
//       )
//     ).toEqual(true);
//   });
// });

// Wait for events to complete
//   setTimeout(() => {
//     expect(
//       wrapper.containsAllMatchingElements([
//         <NavLink to="/tasks/1">Les leksjon</NavLink>,
//         <NavLink to="/tasks/2">Møt opp på forelesning</NavLink>,
//         <NavLink to="/tasks/3">Gjør øving</NavLink>,
//       ])
//     ).toEqual(true);
//     done();
//   });
// });

//   test('TaskNew correctly sets location on create', (done) => {
//     const wrapper = shallow(<TaskNew />);

//     wrapper.find(Form.Input).simulate('change', { currentTarget: { value: 'Kaffepause' } });
//     // @ts-ignore
//     expect(wrapper.containsMatchingElement(<Form.Input value="Kaffepause" />)).toEqual(true);

//     wrapper.find(Button.Success).simulate('click');
//     // Wait for events to complete
//     setTimeout(() => {
//       expect(location.hash).toEqual('#/tasks/4');
//       done();
//     });
//   });
// });
