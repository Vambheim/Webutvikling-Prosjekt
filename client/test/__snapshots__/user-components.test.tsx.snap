// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterUser component tests RegisterUser draws correctly 1`] = `
<Card
  body={false}
  style={
    {
      "border": "none",
      "marginLeft": "auto",
      "marginRight": "auto",
      "padding": "15px",
      "textAlign": "center",
    }
  }
>
  <CardTitle>
    Create user
  </CardTitle>
  <Container
    fluid={false}
    style={
      {
        "marginLeft": "auto",
        "marginRight": "auto",
        "width": "20rem",
      }
    }
  >
    <Row>
      <FormControl
        onChange={[Function]}
        placeholder="Email"
        style={
          {
            "marginBottom": "10px",
            "textAlign": "center",
          }
        }
        type="text"
        value=""
      />
    </Row>
    <Row>
      <FormControl
        onChange={[Function]}
        placeholder="First name"
        style={
          {
            "marginBottom": "10px",
            "textAlign": "center",
          }
        }
        type="text"
        value=""
      />
    </Row>
    <Row>
      <FormControl
        onChange={[Function]}
        placeholder="Last name"
        style={
          {
            "marginBottom": "10px",
            "textAlign": "center",
          }
        }
        type="text"
        value=""
      />
    </Row>
    <Row>
      <FormControl
        onChange={[Function]}
        onKeyUp={[Function]}
        placeholder="Password"
        style={
          {
            "marginBottom": "10px",
            "textAlign": "center",
          }
        }
        type="password"
        value=""
      />
    </Row>
    <Row>
      <FormControl
        onChange={[Function]}
        onKeyUp={[Function]}
        placeholder="Confirm password"
        style={
          {
            "marginBottom": "10px",
            "textAlign": "center",
          }
        }
        type="password"
        value=""
      />
    </Row>
  </Container>
  <Container
    fluid={false}
    style={
      {
        "marginLeft": "auto",
        "marginRight": "auto",
        "width": "15rem",
      }
    }
  >
    <Row>
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        style={
          {
            "marginBottom": "10px",
          }
        }
        variant="success"
      >
        Create user
      </Button>
    </Row>
    <Row>
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        style={
          {
            "marginBottom": "10px",
          }
        }
        variant="outline-secondary"
      >
        Clear input
      </Button>
    </Row>
  </Container>
</Card>
`;

exports[`UserDetails component tests UserDetails draws correctly 1`] = `
<Fragment>
  <Card
    body={false}
    style={
      {
        "marginLeft": "auto",
        "marginRight": "auto",
        "padding": "15px",
        "textAlign": "center",
      }
    }
  >
    <CardTitle>
      User page for  
    </CardTitle>
    <Row
      style={
        {
          "fontSize": "17px",
        }
      }
    >
      <CardText>
        Your email-adress: 
      </CardText>
    </Row>
    <Row
      style={
        {
          "fontSize": "17px",
        }
      }
    >
      <CardText>
        Your name: 
         
      </CardText>
    </Row>
    <Row>
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        style={
          {
            "marginBottom": "10px",
            "marginLeft": "auto",
            "marginRight": "auto",
            "width": "15rem",
          }
        }
        variant="outline-danger"
      >
        Log out
      </Button>
    </Row>
  </Card>
  <Card
    body={false}
    style={
      {
        "border": "none",
        "marginLeft": "auto",
        "marginRight": "auto",
        "padding": "15px",
        "textAlign": "center",
      }
    }
  >
    <CardTitle>
      Your liked recipes
    </CardTitle>
    <Row>
      <Col
        lg={true}
      >
        <Row
          className="g-4"
          md={4}
          xs={1}
        />
      </Col>
    </Row>
  </Card>
</Fragment>
`;

exports[`UserLogIn widgets tests UserLogIn draws correctly 1`] = `
<Card
  body={false}
  style={
    {
      "border": "none",
      "marginLeft": "auto",
      "marginRight": "auto",
      "padding": "15px",
      "textAlign": "center",
    }
  }
>
  <CardTitle>
    Log in
  </CardTitle>
  <Container
    fluid={false}
    style={
      {
        "marginLeft": "auto",
        "marginRight": "auto",
        "width": "20rem",
      }
    }
  >
    <Row>
      <FormControl
        onChange={[Function]}
        placeholder="Email"
        style={
          {
            "marginBottom": "10px",
            "textAlign": "center",
          }
        }
        type="text"
        value=""
      />
    </Row>
    <Row>
      <FormControl
        onChange={[Function]}
        onKeyUp={[Function]}
        placeholder="Password"
        style={
          {
            "marginBottom": "10px",
            "textAlign": "center",
          }
        }
        type="password"
        value=""
      />
    </Row>
  </Container>
  <Container
    fluid={false}
    style={
      {
        "marginLeft": "auto",
        "marginRight": "auto",
        "width": "15rem",
      }
    }
  >
    <Row>
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        style={
          {
            "marginBottom": "10px",
          }
        }
        variant="success"
      >
        Log in
      </Button>
    </Row>
    <Row>
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        style={
          {
            "marginBottom": "10px",
          }
        }
        variant="outline-success"
      >
        No user? Create one here
      </Button>
    </Row>
    <Row>
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        style={
          {
            "marginBottom": "10px",
          }
        }
        variant="outline-secondary"
      >
        Clear input
      </Button>
    </Row>
  </Container>
</Card>
`;
